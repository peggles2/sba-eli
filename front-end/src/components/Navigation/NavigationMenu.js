import React, {Component} from 'react'
import {Link} from 'react-router-dom'
import {Menu} from 'semantic-ui-react'
import NavigationDropdownItem from "./NavigationDropdownItem";
import axios from "axios";
import {Dropdown} from 'semantic-ui-react'

export default class NavigationMenu extends Component {
  state = {
    activeItem: 'dashboard',
    learningPaths: [],
    learningObjectives: [],
    learningEvents: []
  }

  componentDidMount() {
    this.fetchNavigationData();
  }

  fetchNavigationData() {
    const lpUrl = process.env.REACT_APP_SERVICE_HOST +
        "/learning_paths"

    axios.get(lpUrl)
        .then(res => res.data)
        .then(learningPathData => {
          const learningPaths = learningPathData;
          let learningObjectivesData = []
          this.setState({learningPaths})

          learningPaths.map(lp => {

                const loUrl = process.env.REACT_APP_SERVICE_HOST +
                    "/learning_objectives/" +
                    "?course_id=" +
                    lp.id

                axios.get(loUrl)
                    .then(res2 => {
                      learningObjectivesData[lp.id] = res2.data
                    })
                    .then(tmp => {
                      const learningObjectives = learningObjectivesData;
                      this.setState({learningObjectives})
                    })
                    .catch(error => {
                      console.error(error)
                    })
              }
          )
        })
        .catch(error => {
          console.error(error)
        })

  }

  populateLearningPaths() {
    let lp_nav = ''
    if (this.state.learningPaths.length > 0) {
      lp_nav =
          this.state.learningPaths.map(lp =>
              <div>
              <Menu.Item key={lp.id}
                         as={Link}
                         to={`/learning_paths/${lp.id}`}
                  // name={`learning_paths_${lp.id}`}
                  // active={activeItem === `learning_paths_${lp.id}`}
                  // onClick={this.handleItemClick}
                         text={lp.name}>
                {lp.name}
              </Menu.Item>
              {this.populateLearningObjectives(lp.id)}
              </div>
          )
    }
    return lp_nav
  }

  populateLearningObjectives(courseId) {
    console.log("course id " + courseId)
    console.log(this.state.learningObjectives[courseId])
    let lo_nav = ''
    if (this.state.learningObjectives[courseId] && this.state.learningObjectives[courseId].length > 0) {
      lo_nav =
          this.state.learningObjectives[courseId].map(lo =>
              <NavigationDropdownItem key={lo.id} path={`/learning_objectives/${lo.id}`} name={lo.name} docs={this.state.learningEvents} />
          )
    }
    return lo_nav
  }

  handleItemClick = (e, {name}) => this.setState({
    activeItem: name
  })

  render() {

    const {activeItem} = this.state

    return (
        <Menu vertical>

          <Menu.Item as={Link}
                     to='/'
                     name='dashboard'
                     active={activeItem === 'dashboard'}
                     onClick={this.handleItemClick}>
            Dashboard
          </Menu.Item>

          <Menu.Item as={Link}
                     to='/learning_paths'
                     name='learning_paths'
                     active={activeItem === 'learning_paths'}
                     onClick={this.handleItemClick}>
            Learning Paths
          </Menu.Item>
          <Menu.Menu>
            {this.populateLearningPaths()}
          </Menu.Menu>
        </Menu>
    )
  }
}