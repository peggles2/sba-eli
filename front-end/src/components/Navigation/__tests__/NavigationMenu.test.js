import React from 'react';
import ReactDOM from 'react-dom';
import NavigationMenu from '../NavigationMenu';
import { shallow, mount } from 'enzyme';

describe('NavigationMenu', () => {
  it('should render correctly', () => {
    const wrapper = shallow(<NavigationMenu/>);
    
    expect(wrapper).toMatchSnapshot();
  });
});
