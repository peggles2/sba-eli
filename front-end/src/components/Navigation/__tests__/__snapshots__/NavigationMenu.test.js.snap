// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavigationMenu should render correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <NavigationMenu />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <MenuItem
          active={true}
          as={[Function]}
          name="dashboard"
          onClick={[Function]}
          to="/"
        >
          Dashboard
        </MenuItem>,
        <MenuItem
          active={false}
          as={[Function]}
          name="learning_paths"
          onClick={[Function]}
          to="/learning_paths"
        >
          Learning Paths
        </MenuItem>,
        <MenuMenu>
          
        </MenuMenu>,
      ],
      "vertical": true,
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "active": true,
          "as": [Function],
          "children": "Dashboard",
          "name": "dashboard",
          "onClick": [Function],
          "to": "/",
        },
        "ref": null,
        "rendered": "Dashboard",
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "active": false,
          "as": [Function],
          "children": "Learning Paths",
          "name": "learning_paths",
          "onClick": [Function],
          "to": "/learning_paths",
        },
        "ref": null,
        "rendered": "Learning Paths",
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": "",
        },
        "ref": null,
        "rendered": "",
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <MenuItem
            active={true}
            as={[Function]}
            name="dashboard"
            onClick={[Function]}
            to="/"
          >
            Dashboard
          </MenuItem>,
          <MenuItem
            active={false}
            as={[Function]}
            name="learning_paths"
            onClick={[Function]}
            to="/learning_paths"
          >
            Learning Paths
          </MenuItem>,
          <MenuMenu>
            
          </MenuMenu>,
        ],
        "vertical": true,
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "active": true,
            "as": [Function],
            "children": "Dashboard",
            "name": "dashboard",
            "onClick": [Function],
            "to": "/",
          },
          "ref": null,
          "rendered": "Dashboard",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "active": false,
            "as": [Function],
            "children": "Learning Paths",
            "name": "learning_paths",
            "onClick": [Function],
            "to": "/learning_paths",
          },
          "ref": null,
          "rendered": "Learning Paths",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": "",
          },
          "ref": null,
          "rendered": "",
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
